assembly/src/v1/parsers/io_ops.rs

```diff
@@ -163,21 +234,114 @@ mod tests {
         let mut span_ops: Vec<Operation> = Vec::new();
         let op_0 = Token::new("push.0", 0);
         let op_1 = Token::new("push.1", 0);
-        let op_other = Token::new("push.2", 0);
+        let op_dec = Token::new("push.135", 0);
+        let op_hex = Token::new("push.0x7b", 0);
         let expected = vec![
             Operation::Pad,
             Operation::Pad,
             Operation::Incr,
-            Operation::Push(BaseElement::new(2)),
+            Operation::Push(BaseElement::new(135)),
+            Operation::Push(BaseElement::new(123)),
         ];
 
         parse_push(&mut span_ops, &op_0).expect("Failed to parse push.0");
         parse_push(&mut span_ops, &op_1).expect("Failed to parse push.1");
-        parse_push(&mut span_ops, &op_other).expect("Failed to parse push.2");
+        parse_push(&mut span_ops, &op_dec).expect("Failed to parse push of decimal element 123");
+        parse_push(&mut span_ops, &op_hex).expect("Failed to parse push of hex element 0x7b");
 
         assert_eq!(span_ops, expected);
     }
 
+    #[test]
+    fn push_invalid() {
+        // fails when immediate value is invalid or missing
+        let mut span_ops: Vec<Operation> = Vec::new();
+        let param_idx = 0;
+
+        // value missing
+        let op_no_val = Token::new("pushw", param_idx);
+        assert!(parse_push(&mut span_ops, &op_no_val).is_err());
+
+        // invalid value
+        let op_val_invalid = Token::new("push.abc", param_idx);
+        assert!(parse_push(&mut span_ops, &op_val_invalid).is_err());
+
+        // extra value
+        let op_extra_val = Token::new("pushw.0.1", param_idx);
+        assert!(parse_push(&mut span_ops, &op_extra_val).is_err());
+    }
+
+    #[test]
+    fn pushw() {
+        // pushes a word of 4 immediate values in decimal or hexadecimal onto the stack
+        let mut span_ops: Vec<Operation> = Vec::new();
+        let op = Token::new("pushw.1.23.0x1C8.0", 0);
```

Comment:
Not for this PR: now that I'm seeing how this actually looks, I wonder if we should change the format of `pushw` instruction to always take a single hexadecimal value of 32 bytes. Or maybe making this as an additional allowed format. So, for example, possible formats could be:

* `pushw.1.2.3.4`
* `pushw.0x13244...` (32 bytes)

Let's create an issue to think this through later on.

Reply:
I opened an issue here: https://github.com/maticnetwork/miden/issues/44